pkg/lib/config/testdata/config_tests.yaml

```diff
@@ -1184,7 +1184,10 @@ config:
         nameid_format: urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress
         nameid_attribute_pointer: /preferred_username
         acs_urls:
-          - http://localhost:3000/saml-test
+          - http://localhost:3000/acs
+        destination: http://localhost:3000/destination
```

Comment:
```
acs_urls:
- asc_url: http://localhost:3000/acs
- asc_url: http://localhost:3000/acs
   destination: http://localhost:3000/destination
```

Reply:
@louischan-oursky 
After a second thought, if we say allowing multiple acs_url is for allowing the response to be sent through proxies, it seems doesn't quite make sense if a single service provider expect different `destination` value when using different acs_url, as the response should finally reach the same server. So I prefer to keep the current config structure. 

```
  ACSUrl                         Destination
https://proxy1.example.com --> https://example.authgear.cloud
https://proxy2.example.com --> https://example.authgear.cloud
```

Reply:
If we continue this argument, we can say if the developer wants to have a different ASC_URL, they can create another SP.

Reply:
I remember we talked about another way of supporting a SP. That is by setting a URL to the metadata of a SP. Therefore, I think this SP config should mirror the structure of the metadata of a SP. So if the metadata of a SP supports multiple occurence of ACS_URL, then this should be an array. What do you think?

Reply:
Yes. So acs_url is an array, but destination is not?

Reply:
1. For `recipient`, it is ASC_URL by default. If the config has set `recipient`, use config.
2. For `destination`, it is ASC_URL by default. If the config has set `destination`, use config.
3. For `audience`
  - Always include ACS_URL as one audience.
  - If AuthnRequest has audience, check if it is ASC_URL, or `audience` in config. If violated, reject the request.
  - If config has `audience`, output it.

Reply:
Please put this into our spec as well.

Reply:
I found that the library we are using does not allow us to put multiple audience into it, we may need to define our own response struct.

Reply:
@louischan-oursky I would like to copy the whole `schema.go` from `crewjamsaml` into our codebase, so we can modify it easily, what do you think?

Reply:
Feel free to do it.